pallets/protos/src/tests.rs

```diff
@@ -1,381 +1,920 @@
+		ProtosPallet::upload(
+			Origin::signed(signer), 
+			proto.references.clone(), 
+			proto.category.clone(), 
+			proto.tags.clone(), 
+			proto.linked_asset.clone(), 
+			proto.include_cost.map(|cost| Compact::from(cost)), 
+			proto.data.clone(),
+		)
+	}
+
+
+	#[test]
+	fn upload_should_work() {
+		new_test_ext().execute_with(|| {
+			let dd = DummyData::new();
+
+			let block_number = System::block_number(); //@sinkingsugar
+
+			let proto = dd.proto_fragment;
+
+			assert_ok!(upload(dd.account_id, &proto));
+
+			assert!(<Protos<Test>>::contains_key(proto.get_proto_hash())); 
+			
+			let proto_struct = <Protos<Test>>::get(proto.get_proto_hash()).unwrap();
+
+			// I am using `match` to ensure that this test case fails if a new field is ever added to the `Proto` struct
+			// match proto_struct {
+			// 	Proto {
+			// 		block: 1, 
+			// 		patches: Vec::new(), 
+			// 		include_cost: proto.include_cost.map(|cost| Compact::from(cost)),  
+			// 		creator: dd.account_id, 
+			// 		owner: dd.account_id, 
+			// 		references: proto.references, 
+			// 		category: proto.category,
+			// 		tags: proto.tags,
+			// 		metadata: BTreeMap::new(),
+			// 	} => (),
+			// 	// _ => println!("Time to panic!!! Mayday"),
+			// }
+
+			let correct_proto_struct = Proto {
+				block: block_number, 
+				patches: Vec::new(), 
+				include_cost: proto.include_cost.map(|cost| Compact::from(cost)),  
+				creator: dd.account_id, 
+				owner: ProtoOwner::User(dd.account_id), 
+				references: proto.references.clone(), 
+				category: proto.category.clone(),
+				tags: Vec::new(), // proto.tags,
+				metadata: BTreeMap::new(),
+				tickets_info: TicketsInfo::default(),
+			};
+
+			// Ensure that this test case fails if a new field is ever added to the `Proto` struct
+			match proto_struct {
+				s if s == correct_proto_struct => (),
+				_ => panic!("The correct `Proto` struct was not saved in the StorageMap `Protos`"),
+			}
+
+
+			assert!(<ProtosByCategory<Test>>::get(&proto.category).unwrap().contains(&proto.get_proto_hash()));
+			assert!(<ProtosByOwner<Test>>::get(ProtoOwner::User(dd.account_id)).unwrap().contains(&proto.get_proto_hash()));
+
+			let event = <frame_system::Pallet<Test>>::events().pop().expect("Expected at least one EventRecord to be found").event;
+        	assert_eq!(event, mock::Event::from(pallet_protos::Event::Uploaded(proto.get_proto_hash(), proto.get_proto_cid())));
+		});
+	}
+
+	#[test]
+	fn upload_should_not_works_if_proto_hash_exists() {
+		new_test_ext().execute_with(|| {
+			let dd = DummyData::new();
+			let proto = dd.proto_fragment;
+			upload(dd.account_id, &proto);
+			assert_noop!(
+				upload(dd.account_id, &proto), 
+				Error::<Test>::ProtoExists
+			);
+		});
+	}
+
+	#[test]
+	fn upload_should_work_if_user_staked_enough() {
+
+		new_test_ext().execute_with(|| {
+			
+			let dd = DummyData::new();
+
+			let stake = dd.stake;
+	
+			upload(dd.account_id, &stake.proto_fragment);
```

Comment:
All these functions
```rust
upload(....)
lock(....)
link(...)
stake(....)
```
return a Result that should be handled. It is not a problem, I mean, it's just that they create annoying warnings in the compiler.
A quick fix could be to ignore the returned value:
```rust
_ = upload(....)
_ = lock(....)
_ = link(...)
_ = stake(....)
```

Reply:
Shouldn't the result be handled though? At least `unwrap`.